docs/edgeql/path_resolution.rst

```diff
@@ -0,0 +1,177 @@
+.. _ref_eql_path_resolution:
+
+===============
+Path resolution
+===============
+
+Element-wise operations with multiple arguments in EdgeDB are generally applied
+to the :ref:`cartesian product <ref_reference_cardinality_cartesian>` of all
+the input sets.
+
+.. code-block:: edgeql-repl
+
+    db> select {'aaa', 'bbb'} ++ {'ccc', 'ddd'};
+    {'aaaccc', 'aaaddd', 'bbbccc', 'bbbddd'}
+
+In some cases, this works out fine, but in others it doesn't make sense. Take
+this example:
+
+.. code-block:: edgeql
+
+    select User.first_name ++ ' ' ++ User.last_name;
+
+Should the result of this query be every ``first_name`` in your ``User``
+objects concatenated with every ``last_name``? That's probably not what you
+want.
+
+This is why, in cases where multiple element-wise arguments share a common path
+(``User.`` in this example), EdgeDB factors out the common path rather than
+using cartesian multiplication.
+
+.. code-block:: edgeql-repl
+
+    db> select User.first_name ++ ' ' ++ User.last_name;
+    {'Mina Murray', 'Jonathan Harker', 'Lucy Westenra', 'John Seward'}
+
+We assume this is what you want, but if your goal is to get the cartesian
+product, you can accomplish it one of three ways. You could use
+:eql:op:`detached`.
+
+.. code-block:: edgeql-repl
+
+    edgedb> select User.first_name ++ ' ' ++ detached User.last_name;
+    {
+      'Mina Murray',
+      'Mina Harker',
+      'Mina Westenra',
+      'Mina Seward',
+      'Jonathan Murray',
+      'Jonathan Harker',
+      'Jonathan Westenra',
+      'Jonathan Seward',
+      'Lucy Murray',
+      'Lucy Harker',
+      'Lucy Westenra',
+      'Lucy Seward',
+      'John Murray',
+      'John Harker',
+      'John Westenra',
+      'John Seward',
+    }
+
+You could use :ref:`with <ref_eql_with>` to attach a different symbol to
+your set of ``User`` objects.
+
+.. code-block:: edgeql-repl
+
+    edgedb> with U := (select User)
```

Comment:
```suggestion
    edgedb> with U := User
```
You don't need the `select` for aliasing simple expressions. You usually need `select` if you want to use some of the other clauses (`filter` or `order`, etc.).